cypress/utils/apps/query_enhancements/shared.js

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  DatasetTypes,
+  DATASOURCE_NAME,
+  END_TIME,
+  INDEX_PATTERN_WITH_TIME,
+  INDEX_WITH_TIME_1,
+  QueryLanguages,
+  START_TIME,
+  WORKSPACE_NAME,
+} from './constants';
+
+/**
+ * Returns a randomized string
+ * @returns {string}
+ */
+const getRandomString = () => Math.random().toString(36);
+
+/**
+ * Returns a randomized workspace name
+ * @returns {string}
+ */
+export const getRandomizedWorkspaceName = () =>
+  `${WORKSPACE_NAME}-${getRandomString().substring(7)}`;
+
+/**
+ * Returns a randomized datasource name
+ * @returns {string}
+ */
+export const getRandomizedDatasourceName = () =>
+  `${DATASOURCE_NAME}-${getRandomString().substring(0, 18)}`;
+
+/**
+ * Callback for generateAllTestConfigurations
+ * @callback GenerateTestConfigurationCallback
+ * @param {string} dataset - the dataset name
+ * @param {QueryEnhancementDataset} datasetType - the type of the dataset
+ * @param {QueryEnhancementLanguageData} language - the relevant data for the query language to use
+ * @returns {object}
+ */
+
+/**
+ * Returns an array of test configurations for every query language + dataset permutation
+ * @param {GenerateTestConfigurationCallback} generateTestConfigurationCallback - cb function that generates a test case for the particular permutation
+ * @returns {object[]}
```

Comment:
Note: This return type have been updated from `SavedSearchTestConfig[]` to generic `object[]`. The return type of this function is dictated by the output of the `generateTestConfigurationCallback` function.

If JSDoc had full support for generics or helpful TS utilities like `ReturnType`, I would be able to correctly type the return type of this function based on the return type of the callback, but based on my research JSDoc falls short when it comes to generics.

So unfortunately I had to resort to generic `object[]`. One another example of why we should convert our cypress files to be fully Typescript (not even mentioning just how annoying it generally is to type things using JSDoc)